Sources/Controllers/Map/Widgets/OASunriseSunsetWidget.mm

```diff
@@ -369,4 +373,21 @@ - (NSString *) formatNextTime:(NSTimeInterval)nextTime
     return [dateFormatter stringFromDate:nextDate];
 }
 
+- (void) updateCachedLocation
```

Comment:
It is not clear purpose of `updateCachedLocation`? Why cached location even needed in such implementation?
The `updateCachedLocation` should return `BOOL` and if location not changed - do not recalculate sunrise / sunset time!